cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -38,27 +41,24 @@
 /**
  *  Handles creating indexes for a given datastore.
  */
-class IndexCreator {
+public class IndexCreator {
 
-    private final SQLDatabase database;
     private final Datastore datastore;
     private static Random indexNameRandom = new Random();
 
-    private final ExecutorService queue;
+    private final SQLDatabaseQueue queue;
 
     private static final Logger logger = Logger.getLogger(IndexCreator.class.getName());
 
-    public IndexCreator(SQLDatabase database, Datastore datastore, ExecutorService queue) {
+    public IndexCreator(Datastore datastore, SQLDatabaseQueue queue) {
```

Comment:
I don't think we should couple it to an implementation, I not keen on have to cast later on, but I think that can be fixed with some refactoring of the indexing code, by doing the work during construction to figure out if we can use FTS when we create the database rather than running the check each them, so when calling the FTS method to see if it's there we would know ahead of time. Or perhaps moving said check to the Database Queue, since its a property more todo with the database than the datastore itself.